### PR TITLE
chore: Add husky, lint-staged, prettier

### PR DESCRIPTION
**Pull request by vapurrmaid**
In draft until each screen is previewed/verified

## Summary of Changes

- Add husky, lint-staged for a better developer experience
- Add prettier to format all file types (JSON, MD, YAML)
- Format the entire codebase

### CODE REVIEW

File: admin/environment-management/cvms.md [hunk 1]
```diff
@@ -52,46 +53,46 @@ gcloud beta container clusters create "YOUR_NEW_CLUSTER" \
...
+   ```
 
 1. Create your nodegroup (be sure to provide the correct file name):
 
-    ```console
-    eksctl create nodegroup --config-file=coder-node.yaml
-    ```
+   ```console
+   eksctl create nodegroup --config-file=coder-node.yaml
+   ```
```
Comment (vapurrmaid) [inline admin/environment-management/cvms.md:88]: This fixed the look and feel :nerd_face: 

## Before

Screenshot from 2021-03-23 09-50-57


## After

Screenshot from 2021-03-23 09-51-04

## Note

My process for reviewing these lint/format changes are opening the page to a fragment in 2 separate tabs

1. Live: https://coder.com/docs/admin/environment-management/cvms#amazon-web-services-w-eks
2. Preview: https://codercom-lglwr8awa-codercom.vercel.app/docs/admin/environment-management/cvms#amazon-web-services-w-eks

And flip back and forth inspecting where any differences occur.

**Review by jawnsy — APPROVED** [review]
LGTM! :tada: I didn't look too deeply since these changes are mostly automated

File: admin/environment-management/cvms.md (hunk 1 above)
Comment (jawnsy) [reply admin/environment-management/cvms.md:88]: I wonder if we can make a page that iframes both side-by-side and links the vertical scroll... would be really helpful for visual diffs :thinking: Maybe a hackathon project!

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -34,4 +34,6 @@ jobs:
         run: yarn
 
       - name: Lint
```
[the remark this answers is not in the document]
Comment (vapurrmaid) [reply .github/workflows/ci.yaml:36]: I have a question on the 2nd point!

```bash
yarn fmt
```

runs [`prettier --check`](https://prettier.io/docs/en/cli.html#--check) under the hood, which exits with an error code and reports which files are in violation of its rules:

> Checking formatting...
[warn] src/fileA.js
[warn] src/fileB.js
[warn] Code style issues found in the above file(s). Forgot to run Prettier?

Thus, I don't feel it's necessary to check if files have changed (unless I'm not considering and/or missing something). I believe that approach was used in `m/` due to the myraid of languages and different tools such as shellcheck and gofmt etc.

[the remark this answers is not in the document]
Comment (jawnsy) [reply .github/workflows/ci.yaml:36]: Ah, cool! I didn't realize prettier had that option, in the monorepo we're doing `prettier --write` and checking for changes, but this seems better lol

**Pull request merged**